### PR TITLE
Gate emit unit tests on JitDumpEmitUnitTests instead of define

### PR DESCRIPTION
**Pull request by a74nh**
- Add a common `genEmitterUnitTests()`
- Add a config option `JitDumpEmitUnitTests` instead of using `#defines`.
- Add basic string checking to decide which groups to run
- Add a code to jump over the tests to prevent coreclr from running the tests (which usually causes a segfault loading from bad memory)

### CODE REVIEW

**Comment by a74nh** [comment]
@kunalspathak @BruceForstall - as suggested on https://github.com/dotnet/runtime/pull/95918

File: src/coreclr/jit/codegenlinear.cpp [hunk 1]
```diff
@@ -2672,3 +2664,90 @@ void CodeGen::genCodeForSetcc(GenTreeCC* setcc)
...
+// Unit testing of the emitter: generate a bunch of instructions into the last block
+// (it's as good as any, but better than the prologue, which can only be a single instruction
+// group) then use DOTNET_JitLateDisasm=* to see if the late disassembler
+// thinks the instructions are the same as we do.
+&&defined(LATE_DISASM) genAmd64EmitterUnitTests();
+
+genArm64EmitterUnitTests();
+#endif // TARGET_ARM64
+
+/*****************************************************************************
```
Comment (BruceForstall) [inline src/coreclr/jit/codegenlinear.cpp:2689]: This should all be removed?

File: src/coreclr/jit/codegenlinear.cpp [hunk 2]
```diff
@@ -2672,3 +2664,90 @@ void CodeGen::genCodeForSetcc(GenTreeCC* setcc)
...
+ * Unit testing of the emitter: If JitDumpEmitUnitTests is set, generate
+ * a bunch of instructions into the prolog (it's as good a place as any), then:
```
Comment (BruceForstall) [inline src/coreclr/jit/codegenlinear.cpp:2691]:
```suggestion
 * a bunch of instructions, then:
```

**Review by BruceForstall — COMMENTED** [review]
This looks great, except it looks like some "garbage" code to be removed.

As a follow-up, I think we should move all the arm64 unit tests to a separate file (named codegenarm64test.cpp?) since they're now over 5000 lines in this 11000 line file.

File: src/coreclr/jit/codegenlinear.cpp (hunk 1 above)
Comment (a74nh) [reply src/coreclr/jit/codegenlinear.cpp]: Oops. Looks like this was a bad merge. Removed.

**Comment by a74nh** [comment]
> As a follow-up, I think we should move all the arm64 unit tests to a separate file (named codegenarm64test.cpp?) since they're now over 5000 lines in this 11000 line file.

Done.

**Comment by kunalspathak** [comment]
seems like we don't need the `altjit` check, because we should be able to generate the new sve when running natively on arm64?

https://github.com/dotnet/runtime/blob/6d0a902c9585d98bfa44f514bac21a47eabe02fa/src/coreclr/jit/codegenxarch.cpp#L9048-L9052

**Review by kunalspathak — APPROVED** [review]
LGTM. Thanks for doing it.

**Comment by a74nh** [comment]
> seems like we don't need the `altjit` check, because we should be able to generate the new sve when running natively on arm64?
> 
> https://github.com/dotnet/runtime/blob/6d0a902c9585d98bfa44f514bac21a47eabe02fa/src/coreclr/jit/codegenxarch.cpp#L9048-L9052

I originally had this check in the new common function. But then Arm64 testing would always exit without doing the tests. So, I removed it for all targets because it's gated by the config option anyway.

**Pull request merged**